app/forms/api/profile_creation_form.rb

```diff
@@ -0,0 +1,151 @@
+# rubocop:disable IdentityIdp/ErrorsAddLinter
+module Api
+  class ProfileCreationForm
+    include ActiveModel::Model
+
+    validate :valid_user_bundle
+    validate :valid_user
+    validate :valid_password
+
+    attr_reader :user_password, :user_bundle, :user_bundle_headers, :user_bundle_payload
+    attr_reader :user_session
+    attr_reader :pii, :profile, :personal_key
+
+    def initialize(user_password:, user_bundle:, user_session:)
+      @user_password = user_password
+      @user_bundle = user_bundle
+      @user_session = user_session
+      set_idv_session
+    end
+
+    def submit
+      @form_valid = valid?
+
+      if form_valid?
+        create_profile
+        cache_encrypted_pii
+        generate_personal_key
```

Comment:
maybe we do `@personal_key = generate_personal_key` and have a separate `attr_reader` for the personal key? (related to previous comment about sensitive values inside logging payloads)